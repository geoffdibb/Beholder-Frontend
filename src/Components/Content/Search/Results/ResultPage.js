import React from 'react';
import { Form, Col, Row, FormGroup, Label, Input } from 'reactstrap';

import ResultList from './ResultList';

export default class ResultPage extends React.Component {

    searchdb = (e) => {
        e.preventDefault();
        let category = e.target[0].value;
        if (e.target[1].value === "") {
            this.setState({ message: "Please fill all fields" });
        } else {
            this.props.search(category, e.target[1].value);
        }
    }
    render() {
        return (
            <div>
                <h2>Results</h2>
                <Form onSubmit={this.searchdb}>
                    <Row style={{ display: 'flex', justifyContent: 'center', alignItems: 'center' }}>
                        <div style={{ display: 'flex', justifyContent: 'center', alignItems: 'center' }}>
                            <FormGroup>
                                <Label for="Category" hidden>Category</Label>
                                <Input type="select">
                                    <option>Name</option>
                                    <option>Location</option>
                                    <option>Car Reg</option>
                                    <option>Time</option>
                                </Input>
                            </FormGroup>


                            <FormGroup>
                                <Label for="Search" hidden>Search</Label>
                                <Input type="search" name="search" id="Search" placeholder="Search" />

                            </FormGroup>

                        </div>
                    </Row>
                    <Row>
                        <Col md='4'></Col>
                        <Col md='4'>
                            <input type="submit" className="btn btn-dark" value="Search" />
                        </Col>
                        <Col md='4'></Col>
                    </Row>
                </Form>
                <br></br>
                <Row style={{ justifyContent: 'center', alignItems: 'center' }}>
                    <Col md='8'>
                        <ResultList searchResults={this.props.searchResults} search={this.props.search} selectProfile={this.props.selectProfile} />
                    </Col>

                </Row>
            </div>
        );
    };

}