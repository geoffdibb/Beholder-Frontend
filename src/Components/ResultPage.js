import React from 'react';

export default class ResultPage extends React.Component {

    render() {
        return (
            <div>
            </div>
        );
    };

}