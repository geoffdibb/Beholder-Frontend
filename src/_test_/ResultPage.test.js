import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import ResultPage from '../Components/Content/Search/Results/ResultPage';

import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer';


Enzyme.configure({ adapter: new Adapter() })

it('snapshot results', () => {
    const tree = renderer.create(<ResultPage/>).toJSON()
    expect(tree).toMatchSnapshot();
  })