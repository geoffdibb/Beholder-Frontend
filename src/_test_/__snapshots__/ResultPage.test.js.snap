// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot results 1`] = `
<div>
  <h2>
    Search
  </h2>
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="row"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
          }
        }
      >
        <div
          className="form-group"
        >
          <label
            className="sr-only"
            htmlFor="Category"
          >
            Category
          </label>
          <select
            className="form-control"
          >
            <option>
              Forenames
            </option>
            <option>
              Surname
            </option>
            <option>
              Car Reg
            </option>
          </select>
        </div>
        <div
          className="form-group"
        >
          <label
            className="sr-only"
            htmlFor="Search"
          >
            Search
          </label>
          <input
            className="form-control"
            id="Search"
            name="search"
            placeholder="Search"
            type="search"
          />
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-4"
      />
      <div
        className="col-md-4"
      >
        <input
          className="btn btn-dark"
          type="submit"
          value="Search"
        />
      </div>
      <div
        className="col-md-4"
      />
    </div>
  </form>
  <br />
  <div
    className="row"
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <div
      className="col-md-8"
    >
      <div>
        <div />
      </div>
    </div>
  </div>
</div>
`;
